src/test/kotlin/com/hibob/UserServiceTest.kt

```diff
@@ -0,0 +1,99 @@
+package com.hibob
+
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.kotlin.*
+
+class UserServiceTest {
+
+
+    private val userDao: UserDao = mock<UserDao>{}
+    private val notificationService: NotificationService = mock<NotificationService>{}
+    private val emailVerificationService: EmailVerificationService = mock<EmailVerificationService>{}
+    private val userService = UserService(userDao, notificationService, emailVerificationService)
+
+    @Test
+    fun `registerUser should throw if user id exist`() {
+        val user = User(1, "a", "a@a.co", "1q2w3e", false)
+
+        whenever(userDao.findById(user.id)).thenReturn(user)
+        assertThrows<IllegalArgumentException>("User already exists") { userService.registerUser(user) }
+    }
+
+    @Test
+    fun `registerUser - registerUser throw if email verified`() {
+        val user = User(1, "a", "a@a.co", "1q2w3e", false)
+        whenever(userDao.findById(user.id)).thenReturn(null)
+        whenever(userDao.save(user.copy(isEmailVerified = false))).thenReturn(false)
+        assertThrows<IllegalStateException>("\"User registration failed\"") { userService.registerUser(user)}
```

Comment:
In order to test the exception message you need to save the returned value from the assertThrows

        val exception = assertThrows(IllegalArgumentException::class.java) {
            userService.registerUser(user)
        }